cmd/serf/command/agent/config.go

```diff
@@ -127,6 +127,9 @@ type Config struct {
 	// are "wan", "lan", and "local". The default is "lan"
 	Profile string `mapstructure:"profile"`
 
+	// ProfilePath is path to .json file for configuration memberlist
+	ProfilePath string `mapstructure:"profile_path"`
```

Comment:
What's the benefit to putting this in a separate json file instead of making it just a section of the config?

Reply:
I was wondering about that, but simply migrated the existing work which @slackpad had already reviewed without changing it too much. I'm tempted to suggest that these values should be merged into the main `Config` structure instead such that they can be placed in _any_ JSON file in the configuration directory?

Reply:
Yeah that's what I was thinking, replacing ProfilePath with a Profile struct seems easier to use. The only downside would be not being able to configure these things via command line, which is probably fine.

Reply:
I think that's a worthwhile change. There is a related issue about doing the same for the key ring elsewhere, though that can be done as a separate step.